reports/helmcharts/helm-chart-versions.sh

```diff
@@ -130,21 +129,47 @@ for chart in $(echo "$charts" | jq -c '.[]'); do
   fi
 
   # Enhance document with additional information
+  helm_chart_name=$(echo "$chart" | jq -r '.name')
   uuid=$(uuidgen)
   created_on=$(date '+%Y-%m-%d %H:%M:%S')
+  id="${cluster_name}-${helm_chart_name}"
+
 
   document=$(echo "$chart" | jq --arg cluster_name "$cluster_name" \
                                 --arg verdict $verdict \
-                                --arg id "$uuid" \
+                                --arg id "$id" \
                                 --arg environment "$environment" \
                                 --arg created_on "$created_on" \
                                 --arg report_type "table" \
                                 --arg display_name "HELM Repositories" \
                                 --arg color_code $color_code '. + {id: $id, environment: $environment, createdOn: $created_on, lastUpdated: $created_on, displayName: $display_name, cluster: $cluster_name, verdict: $verdict, colorCode: $color_code, reportType: $report_type}')
 
   documents+=("$document")
+  
 done
 
+cosmosdb_account_name="sds-platform-version-reporter"
+cosmosdb_database_name="reports"
+cosmosdb_container_name="helmcharts"
+id_to_check="$id"
+
+query_result=$(az cosmosdb sql container execute-query \
+--account-name "$cosmosdb_account_name" \
+--database-name "$cosmosdb_database_name" \
+--name "$cosmosdb_container_name" \
+--query "SELECT * FROM c WHERE c.id = '$id_to_check'" \
+--output json)
+
+if [[ $query_result == "[]" ]]; then
```

Comment:
You are checking the query to see if the Id exists which is fine but what happens if the versions change, we'd never get an updated record in Cosmos with this method.
We need to lookup the record and if it exists then we need to check the verdict matches.
Lines 106-129 set the verdict value for the script and if that is different to whats stored in Cosmos then we need to update it even though the record/Id already exist.